Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM golang:1.17 as builder
+
+ARG GOPROXY=https://goproxy.io
+
+ENV GOPROXY=${GOPROXY}
```

Comment:
The name of the ARG and the ENV is the same; it might not be the clearest way.

How about `ARG GOPROXY_ARG`, and `ENV GOPROXY`?

Reply:
Ok, xx_ARG is clear

Reply:
> I'm having this error locally (with GOPROXY, without VPN, without changing anything from this PR, M1 Mac):
> 
> ```shell
> tiexin@MBA ~/work/hutao/stream $ make build-with-docker
> rm -rf build/ && mkdir build
> # update the deps layer if needed
> docker build --build-arg GOPROXY=https://goproxy.io -t devstream_build:v0.1 .
> [+] Building 103.8s (9/9) FINISHED                                                                                                                                                                                        
>  => [internal] load build definition from Dockerfile                                                                                                                                                                 0.0s
>  => => transferring dockerfile: 311B                                                                                                                                                                                 0.0s
>  => [internal] load .dockerignore                                                                                                                                                                                    0.0s
>  => => transferring context: 2B                                                                                                                                                                                      0.0s
>  => [internal] load metadata for docker.io/library/golang:1.17                                                                                                                                                       0.0s
>  => [1/4] FROM docker.io/library/golang:1.17                                                                                                                                                                         0.0s
>  => [internal] load build context                                                                                                                                                                                    0.0s
>  => => transferring context: 55B                                                                                                                                                                                     0.0s
>  => CACHED [2/4] WORKDIR /devstream                                                                                                                                                                                  0.0s
>  => CACHED [3/4] COPY go.mod go.sum /devstream/                                                                                                                                                                      0.0s
>  => [4/4] RUN go mod download                                                                                                                                                                                       90.6s
>  => exporting to image                                                                                                                                                                                              13.2s
>  => => exporting layers                                                                                                                                                                                             13.1s
>  => => writing image sha256:6a06db6b4aa8e94ade6e7a6b7dae008dfabb4e68c8902c04fb2f214583243ecb                                                                                                                         0.0s
>  => => naming to docker.io/library/devstream_build:v0.1                                                                                                                                                              0.0s
> 
> Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
> cp -r go.mod go.sum cmd internal build.sh build/
> docker run -v /Users/tiexin/work/hutao/stream/build:/devstream -e GOOS=linux -e GOARCH=amd64 --rm devstream_build:v0.1
> # github.com/merico-dev/stream/cmd/githubactions
> loadinternal: cannot find runtime/cgo
> /usr/local/go/pkg/tool/linux_arm64/link: running gcc failed: exit status 1
> gcc: error: unrecognized command-line option '-m64'
> 
> make: *** [build-with-docker] Error 2
> ```
> 
> I think the first error `loadinternal: cannot find runtime/cgo` is because of missing `CGO_ENABLED=1` when building, but the second error `gcc: error: unrecognized command-line option '-m64'` I've got no clue. I bet it's related to cross-platform building, but no matter how I changed the GOOS and GOARCH, it was more or less the same.
> 
> @daniel-hutao please have a look.

It does well at my mac(x86_64), I will see if it can be reproduced after getting the new Mac with m1.